rbac/management/group/relation_api_dual_write_subject_handler.py

```
@@ -478,3 +478,46 @@ def _migrate_custom_role(self, role):
+        for role in roles:
+            if role.system:
+                system.append(role)
+            else:
+                custom.append(role)
+
+        if len(system) > 0:
+            id_params = ", ".join(["%s"] * len(system))
+
+            locked_system = list(
+                Role.objects.raw(
+                    f"SELECT * FROM management_role WHERE system = TRUE AND id IN ({id_params}) FOR SHARE",
```

Ruff (0.16.0)

[error] Possible SQL injection vector through string-based query construction

(S608)